File: src/main/java/cz/vhromada/catalog/gui/program/ProgramsListDataModel.java
package cz.vhromada.catalog.gui.program;

import java.util.List;

import cz.vhromada.catalog.entity.Program;
import cz.vhromada.catalog.facade.ProgramFacade;
import cz.vhromada.catalog.gui.commons.AbstractListDataModel;
import cz.vhromada.result.Result;

import org.springframework.util.Assert;

/**
 * A class represents data model for list with programs.
 *
 * @author Vladimir Hromada
 */
public class ProgramsListDataModel extends AbstractListDataModel<Program> {

    /**
     * SerialVersionUID
     */
    private static final long serialVersionUID = 1L;

    /**
     * Facade for programs
     */
    private ProgramFacade programFacade;

    /**
     * Creates a new instance of ProgramsListDataModel.
     *
     * @param programFacade facade for programs
     * @throws IllegalArgumentException if facade for programs is null
     */
    public ProgramsListDataModel(final ProgramFacade programFacade) {
        Assert.notNull(programFacade, "Facade for programs mustn't be null");

        this.programFacade = programFacade;
        update();
    }

    @Override
    protected Result<List<Program>> getData() {
        return programFacade.getAll();
    }

    @Override
    protected String getDisplayValue(final Program dataObject) {
        return dataObject.getName();
    }

}

File: src/main/kotlin/cz/vhromada/catalog/gui/program/ProgramsListDataModel.kt
package cz.vhromada.catalog.gui.program

import cz.vhromada.catalog.entity.Program
import cz.vhromada.catalog.facade.ProgramFacade
import cz.vhromada.catalog.gui.common.AbstractListDataModel
import cz.vhromada.common.result.Result

/**
 * A class represents data model for list with programs.
 *
 * @author Vladimir Hromada
 */
class ProgramsListDataModel(private val programFacade: ProgramFacade) : AbstractListDataModel<Program>() {

    init {
        update()
    }

    override fun getData(): Result<List<Program>> {
        return programFacade.getAll()
    }

    override fun getDisplayValue(dataObject: Program): String {
        return dataObject.name!!
    }

}
